pwchem/protocols/VirtualDrugScreening/protocol_scorch2_pose.py

```diff
@@ -0,0 +1,292 @@
+# -*- coding: utf-8 -*-
+# **************************************************************************
+# *
+# * Authors: Blanca Pueche (blanca.pueche@cnb.csic.es)
+# *
+# * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC
+# *
+# * This program is free software; you can redistribute it and/or modify
+# * it under the terms of the GNU General Public License as published by
+# * the Free Software Foundation; either version 2 of the License, or
+# * (at your option) any later version.
+# *
+# * This program is distributed in the hope that it will be useful,
+# * but WITHOUT ANY WARRANTY; without even the implied warranty of
+# * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# * GNU General Public License for more details.
+# *
+# * You should have received a copy of the GNU General Public License
+# * along with this program; if not, write to the Free Software
+# * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+# * 02111-1307  USA
+# *
+# *  All comments concerning this program package may be sent to the
+# *  e-mail address 'you@yourinstitution.email'
+# *
+# **************************************************************************
+
+
+"""
+SCORCH2 (SC2) is a machine learning rescoring model designed for interaction-based virtual screening. (SC2, https://github.com/LinCompbio/SCORCH2)
+
+"""
+import csv
+import logging
+from ctypes.wintypes import SMALL_RECT
+from pathlib import Path
+
+from pyworkflow.protocol import params
+from pwem.protocols import EMProtocol
+
+from pwchem.objects import SmallMolecule, SetOfSmallMolecules
+from pwchem.utils import os, shutil, re, runOpenBabel
+from pwchem import Plugin, SCORCH2_DIC
+
+
+currentDir = Path(__file__).parent.resolve()
+
+
+
+class ProtocolSCORCH2(EMProtocol):
+    """
+    Computes the best poses of protein-ligand interactions using SCORCH2: https://github.com/LinCompbio/SCORCH2
+    """
+    _label = 'SCORCH2 rescoring'
+
+    # -------------------------- DEFINE param functions ----------------------
+    def _defineParams(self, form):
+        form.addSection(label='Input')
+        iGroup = form.addGroup('Input')
+        # Pre-extracted features
+        iGroup.addParam('useFeatures', params.BooleanParam, default=False,
+                       label="Use extracted features: ",
+                       help='Choose whether to use pre-extracted features directly.')
+        iGroup.addParam('inputFeatures', params.PathParam,
+                        label='Input features file: ',
+                        condition='useFeatures',
+                        help='Input file where the pre-extracted features are stored.')
+        # Ligand and protein file
+        iGroup.addParam('inputPDBproteinFile', params.PointerParam, pointerClass='AtomStruct',
```

Comment:
The AtomStruct object the ligands are docked to is linked to the SetOfSmallMolecules object, so it is not needed as a separated input. You can validate that the input SetOfMols has indeed this _proteinFile attribute to confirm that the protocol can be run